src/components/ArticleList.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState, useContext } from "react";
+import { getArticles } from "../api";
+import { ArticleCard } from "./ArticleCard";
+import Grid from "@mui/material/Grid";
+
+export const ArticleList = () => {
+  const [articles, setArticles] = useState([]);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    console.log("use effect running");
+    setIsLoading(true);
+    getArticles()
+      .then(({ data }) => {
+        console.log(data, "<<<<data in list");
+        setArticles(data.articles);
+        setIsLoading(false);
+        setError(false);
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  }, []);
+
+  if (articles) {
```

Comment:
Instead of if articles here I would have an if for if isLoading, if it it's true render a loading message. 

I would also do the same for error.

Once you have those you won't need to check if there are articles, because you will always get either loading, error or articles.